proposals/lip-0017.md

```diff
@@ -89,105 +93,133 @@ There are, however, still some limitations that disallow arbitrarily large numbe
+2. The `asset` property needs to contain the property `optionalKeys`. The value of this property is an array of pairwise distinct public keys. This array must be ordered lexicographically. Once the account is registered as a multisignature account, every outgoing transaction requires some signatures for some public keys in `optionalKeys` (the number of needed signatures depends on the `numberOfSignatures` property and may also be zero). The array may be empty.
+3. The sets of mandatory and optional keys, specified by `mandatoryKeys` and `optionalKeys,` must be disjoint; their union must have at least one element and at most 64 elements.
 4. The `asset` property needs to contain the `numberOfSignatures` property.
-    - The value must be an integer between 1 and the total number of public keys of the account (sum of optional and mandatory keys).
-    - The value must not be smaller than the length of the value of `mandatoryKeys`.
-    - The value specifies how many signatures are needed for every outgoing transaction from the registered multisignature account, where signatures for mandatory and optional keys are counted.
-5. The transaction object needs to contain the `signatures` property. The value of this property is an array where each entry is an object with the properties `index` and `signature`. The value of index must be an integer. Moreover:
-    - For every entry `e`, the signature `e.signature` must be a valid signature of the registration transaction for the public key of the account with index `e.index` (see [below](#indexes-of-public-keys) for the specification of _index_).
-    - For any two distinct entries `e1` and `e2`, the indexes `e1.index` and `e2.index` must be distinct.
-    - For every public key in `mandatoryKeys` or `optionalKeys,` there must be an entry with a valid signature and matching index.
-    - If the sender public key is also contained in one of the two arrays, then there must be a index-signature entry for this public key as well.
+   - The value must be an integer between 1 and the total number of public keys of the account (sum of optional and mandatory keys).
+   - The value must not be smaller than the length of the value of `mandatoryKeys`.
+   - The value specifies how many signatures are needed for every outgoing transaction from the registered multisignature account, where signatures for mandatory and optional keys are counted.
+5. The `signatures` property of the transaction object is appended with signatures corresponding to the public keys contained in `mandatoryKeys` and `optionalKeys`. All public keys must have a corresponding signature. The signatures corresponding to `mandatoryKeys` are appended first, followed by those corresponding to `optionalKeys`, where the order of signatures is the same as the order of public keys in the respective array.
```

Comment:
In the case of `optionalKeys` the order might be the same but gaps can occur so it would not be possible to pair signatures to keys. For example if we have as `optionalKeys` -> `{ a, b, c }` a transaction where with signatures from `{a, c}` - signatures { 0xab..., 0xcd...} - might be sent and they might be ordered but how do we check that `0xcd` corresponds to key `c` instead of `b`?

Reply:
For the registration transaction, there needs to be a signature for every public key. So, no gaps in this case.